packages/grid/_modules_/grid/components/column-header-filter-icon.tsx

```diff
@@ -0,0 +1,232 @@
+//   paper: {
+//     maxWidth: 400,
+//     overflow: 'auto',
+//   },
+//   popper: {
+//     zIndex: 1,
+//     '&[x-placement*="bottom"] $arrow': {
+//       top: 0,
+//       left: 0,
+//       marginTop: '-0.9em',
+//       width: '3em',
+//       height: '1em',
+//       '&::before': {
+//         borderWidth: '0 1em 1em 1em',
+//         borderColor: `transparent transparent ${theme.palette.background.paper} transparent`,
+//       },
+//     },
+//     '&[x-placement*="top"] $arrow': {
+//       bottom: 0,
+//       left: 0,
+//       marginBottom: '-0.9em',
+//       width: '3em',
+//       height: '1em',
+//       '&::before': {
+//         borderWidth: '1em 1em 0 1em',
+//         borderColor: `${theme.palette.background.paper} transparent transparent transparent`,
+//       },
+//     },
+//     '&[x-placement*="right"] $arrow': {
+//       left: 0,
+//       marginLeft: '-0.9em',
+//       height: '3em',
+//       width: '1em',
+//       '&::before': {
+//         borderWidth: '1em 1em 1em 0',
+//         borderColor: `transparent ${theme.palette.background.paper} transparent transparent`,
+//       },
+//     },
+//     '&[x-placement*="left"] $arrow': {
+//       right: 0,
+//       marginRight: '-0.9em',
+//       height: '3em',
+//       width: '1em',
+//       '&::before': {
+//         borderWidth: '1em 0 1em 1em',
+//         borderColor: `transparent transparent transparent ${theme.palette.background.paper}`,
+//       },
+//     },
+//   },
+//   arrow: {
+//     position: 'absolute',
+//     fontSize: 7,
+//     width: '3em',
+//     height: '3em',
+//     '&::before': {
+//       content: '""',
+//       margin: 'auto',
+//       display: 'block',
+//       width: 0,
+//       height: 0,
+//       borderStyle: 'solid',
+//     },
+//   },
+// }));
+
+export const ColumnFilterMenu: React.FC = () => {
+  // const classes = useStyles();
+  const apiRef = React.useContext(ApiContext);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [filterValue, setFilterValue] = React.useState('');
+  const [target, setTarget] = React.useState<HTMLElement | null>(null);
+  const [colDef, setColumn] = React.useState<ColDef | null>(null);
+  const icons = useIcons();
+  const filterIconElement = React.createElement(icons.columnFiltering!, {});
+
+  const hideTimeout = React.useRef<any>();
+  const hidePopper = React.useCallback(() => {
+    hideTimeout.current = setTimeout(() => setIsOpen(() => false), 50);
+  }, []);
+
+  const onColumnFilterClick = React.useCallback(
+    ({ element, column }) => {
+      setImmediate(() => clearTimeout(hideTimeout.current));
+      setIsOpen((p) => {
+        if (colDef == null || column.field !== colDef?.field) {
+          setFilterValue('');
+          return true;
+        }
+        return !p;
+      });
+      setTarget(element);
+      setColumn(column);
+    },
+    [colDef],
+  );
+
+  const onFilterChange = React.useCallback((event) => {
```

Comment:
TextField doesn't use memo. I believe that the best way to handle performance is to only consider it once an issue Is **measured**. Why bother using `useCallback` here?

```suggestion
  const onFilterChange = (event) => {
```